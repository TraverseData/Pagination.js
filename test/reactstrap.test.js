
import React from 'react'
import renderer from 'react-test-renderer'

import Pagination from '../react/PaginationReactstrap'


test('1st page, 100 total, 10 per page', () => {
    const component = renderer.create(
        <Pagination
            page={1}
            count={100}
            perPage={10}
            href={page => page}
        />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})

test('3rd page, 45 total, 10 per page', () => {
    const component = renderer.create(
        <Pagination
            page={3}
            count={45}
            perPage={10}
            href={page => page}
        />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})

test('10th page, 200 total, 10 per page', () => {
    const component = renderer.create(
        <Pagination
            page={10}
            count={200}
            perPage={10}
            href={page => page}
        />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})

test('10th page, 100 total, 10 per page', () => {
    const component = renderer.create(
        <Pagination
            page={10}
            count={100}
            perPage={10}
            href={page => page}
        />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})

test('1st page, 10 total, 10 per page', () => {
    const component = renderer.create(
        <Pagination
            page={1}
            count={10}
            perPage={10}
            href={page => page}
        />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})
