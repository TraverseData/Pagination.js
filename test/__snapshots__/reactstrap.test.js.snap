// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react/PaginationReactstrap 1st page, 10 total, 10 per page 1`] = `null`;

exports[`react/PaginationReactstrap 1st page, 100 total, 10 per page 1`] = `
<ul
  className="pagination"
>
  <li
    className="page-item active"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={1}
      onClick={undefined}
      role="button"
    >
      1
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={2}
      onClick={undefined}
      role="button"
    >
      2
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={3}
      onClick={undefined}
      role="button"
    >
      3
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
    >
      ...
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={10}
      onClick={undefined}
      role="button"
    >
      10
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={2}
      onClick={undefined}
      role="button"
    >
      Next
    </a>
  </li>
</ul>
`;

exports[`react/PaginationReactstrap 3rd page, 45 total, 10 per page 1`] = `
<ul
  className="pagination"
>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={2}
      onClick={undefined}
      role="button"
    >
      Previous
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={1}
      onClick={undefined}
      role="button"
    >
      1
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={2}
      onClick={undefined}
      role="button"
    >
      2
    </a>
  </li>
  <li
    className="page-item active"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={3}
      onClick={undefined}
      role="button"
    >
      3
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={4}
      onClick={undefined}
      role="button"
    >
      4
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={5}
      onClick={undefined}
      role="button"
    >
      5
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={4}
      onClick={undefined}
      role="button"
    >
      Next
    </a>
  </li>
</ul>
`;

exports[`react/PaginationReactstrap 10th page, 100 total, 10 per page 1`] = `
<ul
  className="pagination"
>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={9}
      onClick={undefined}
      role="button"
    >
      Previous
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={1}
      onClick={undefined}
      role="button"
    >
      1
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
    >
      ...
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={8}
      onClick={undefined}
      role="button"
    >
      8
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={9}
      onClick={undefined}
      role="button"
    >
      9
    </a>
  </li>
  <li
    className="page-item active"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={10}
      onClick={undefined}
      role="button"
    >
      10
    </a>
  </li>
</ul>
`;

exports[`react/PaginationReactstrap 10th page, 200 total, 10 per page 1`] = `
<ul
  className="pagination"
>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={9}
      onClick={undefined}
      role="button"
    >
      Previous
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={1}
      onClick={undefined}
      role="button"
    >
      1
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
    >
      ...
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={8}
      onClick={undefined}
      role="button"
    >
      8
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={9}
      onClick={undefined}
      role="button"
    >
      9
    </a>
  </li>
  <li
    className="page-item active"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={10}
      onClick={undefined}
      role="button"
    >
      10
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={11}
      onClick={undefined}
      role="button"
    >
      11
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={12}
      onClick={undefined}
      role="button"
    >
      12
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
    >
      ...
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={20}
      onClick={undefined}
      role="button"
    >
      20
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={11}
      onClick={undefined}
      role="button"
    >
      Next
    </a>
  </li>
</ul>
`;

exports[`react/PaginationReactstrap passes other props 1`] = `
<ul
  className="pagination pagination-lg"
>
  <li
    className="page-item active"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={1}
      onClick={undefined}
      role="button"
    >
      1
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={2}
      onClick={undefined}
      role="button"
    >
      2
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={3}
      onClick={undefined}
      role="button"
    >
      3
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
    >
      ...
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={10}
      onClick={undefined}
      role="button"
    >
      10
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      aria-label={undefined}
      className="page-link"
      href={2}
      onClick={undefined}
      role="button"
    >
      Next
    </a>
  </li>
</ul>
`;
